### ISSUE TITLE
Pulling images from outside /Photos directory

### ISSUE THREAD

**Issue by jeffh2os** [nextcloud/photos#1719]
**Describe the bug**
The Photos app is cluttered with a mess of images from documents, presentations and other images not in the /Photos path.

**To Reproduce**
Photos app should only have images from the /Photos directory

**Expected behavior**
I have four directories in /Photos and I have issues finding the photos due to all the clutter the app is pulling in from all my work documents. It should only be the pictures I've put in the /Photos directory.

**Screenshots**
![image](https://user-images.githubusercontent.com/94427932/228878457-e46bcd72-2559-423e-bd7f-1b9e159680b5.png)
(none of the images in this screenshot reside in the /Photos directory)

**Desktop (please complete the following information):**
 - OS: Linux/Windows
 - Browser Firefox/Chrome
 - Version Nextcloud 25.0.3

**Smartphone (please complete the following information):**
 - Device: none
 - OS: none
 - Browser none
 - Version none

**Browser log**
```
TypeError: e.user is undefined
[client.js:1:50171](moz-extension://736682c1-18a7-4f37-a3b5-f7e095e554f4/js/client.js)
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). [contentAllDocStart.bundle.js:1:1763](moz-extension://279de060-c9fc-47c1-ac1b-0e3d49a89ffa/contentAllDocStart.bundle.js)
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). [preload.js:136:52](moz-extension://79dd8f57-5306-4f60-a2b4-a339d33942cb/lib/content-script/preload.js)
Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”). sandbox eval code:9:350
Source map error: Error: request failed with status 403
Resource URL: moz-extension://279de060-c9fc-47c1-ac1b-0e3d49a89ffa/contentAllDocStart.bundle.js


Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///nextcloud/node_modules/@nextcloud/auth/node_modules/@nextcloud/event-bus/dist/index.es.js)
JQMIGRATE: Migrate is installed, version 3.4.0 [jquery-migrate.min.js:2:698](webpack:///nextcloud/node_modules/jquery-migrate/dist/jquery-migrate.min.js)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///nextcloud/node_modules/@nextcloud/auth/node_modules/@nextcloud/event-bus/dist/index.es.js)
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///nextcloud/node_modules/@nextcloud/auth/node_modules/@nextcloud/event-bus/dist/index.es.js)
OCA.Files.Sidebar initialized [Sidebar.js:36:10](webpack:///nextcloud/apps/files/src/services/Sidebar.js)
OCA.Files.Sidebar initialized [sidebar.js:39:8](webpack:///nextcloud/apps/files/src/sidebar.js)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///nextcloud/node_modules/@nextcloud/auth/node_modules/@nextcloud/event-bus/dist/index.es.js)
Proxying an event bus of version 2.1.1 with 1.3.0 [major.js:2](webpack:///node_modules/semver/functions/major.js)
Proxying an event bus of version 2.1.1 with 3.0.0 [index.js:19:12](webpack:///nextcloud/node_modules/@nextcloud/vue/node_modules/@nextcloud/event-bus/dist/index.js)
OCA.Sharing.ShareSearch initialized [ShareSearch.js:33:10](webpack:///nextcloud/apps/files_sharing/src/services/ShareSearch.js)
OCA.Sharing.ExternalLinkActions initialized [ExternalLinkActions.js:33:10](webpack:///nextcloud/apps/files_sharing/src/services/ExternalLinkActions.js)
OCA.Sharing.ExternalShareActions initialized [ExternalShareActions.js:33:10](webpack:///nextcloud/apps/files_sharing/src/services/ExternalShareActions.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [index.esm.js:11:20](webpack:///activity/node_modules/@nextcloud/event-bus/dist/index.esm.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [index.js:19:20](webpack:///activity/node_modules/@nextcloud/event-bus/dist/index.js)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///activity/node_modules/@nextcloud/logger/node_modules/@nextcloud/event-bus/dist/index.es.js)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///nextcloud/node_modules/@nextcloud/auth/node_modules/@nextcloud/event-bus/dist/index.es.js)
Proxying an event bus of version 2.1.1 with 3.0.0 [index.js:19:12](webpack:///nextcloud/node_modules/@nextcloud/vue/node_modules/@nextcloud/event-bus/dist/index.js)
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)
OCA.Comments.View initialized [comments-app.js:32:8](webpack:///nextcloud/apps/comments/src/comments-app.js)
Handlebars is deprecated: please ship your own, this will be removed in Nextcloud 20 2 [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [jsonp chunk loading:1:58](webpack:///viewer/webpack/runtime/jsonp%20chunk%20loading)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///viewer/node_modules/@nextcloud/logger/node_modules/@nextcloud/event-bus/dist/index.es.js)
[WARN] viewer: Some mimes were ignored because they are not enabled in the server previews config 
Object { level: 2, app: "viewer", uid: "xxxx", ignoredMimes: (3) […] }
[ConsoleLogger.js:60:18](webpack:///viewer/node_modules/@nextcloud/logger/dist/ConsoleLogger.js)
OCA.Viewer initialized [Viewer.js:51:25](webpack:///viewer/src/services/Viewer.js)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///richdocuments/node_modules/@nextcloud/vue/node_modules/@nextcloud/auth/node_modules/@nextcloud/event-bus/dist/index.es.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [index.esm.js:11:20](webpack:///event-bus/dist/index.esm.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [index.js:19:20](webpack:///text/node_modules/@nextcloud/event-bus/dist/index.js)
oc_appswebroots is deprecated: use OC.appswebroots instead, this will be removed in Nextcloud 20 [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [index.esm.js:11:12](webpack:///deck/node_modules/@nextcloud/event-bus/dist/index.esm.js)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:6](webpack:///deck/node_modules/@nextcloud/logger/node_modules/@nextcloud/event-bus/dist/index.es.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [index.js:19:12](webpack:///deck/node_modules/@nextcloud/event-bus/dist/index.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [ProxyBus.ts:16:12](webpack:///lib/ProxyBus.ts)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:6](webpack:///photos/node_modules/@nextcloud/logger/node_modules/@nextcloud/event-bus/dist/index.es.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [ProxyBus.ts:16:12](webpack:///lib/ProxyBus.ts)
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 2 [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [ProxyBus.ts:16:20](webpack:///lib/ProxyBus.ts)
Proxying an event bus of version 2.1.1 with 3.0.2 [ProxyBus.ts:16:20](webpack:///lib/ProxyBus.ts)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///notifications/node_modules/@nextcloud/notify_push/node_modules/@nextcloud/auth/node_modules/@nextcloud/event-bus/dist/index.es.js)
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)
Registering notifications container as a menu [NotificationsApp.vue:220](webpack:///notifications/src/NotificationsApp.vue)
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 2 [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)
Notifications permissions granted [NotificationsApp.vue:461](webpack:///notifications/src/NotificationsApp.vue)
Polling interval updated to 30000 [NotificationsApp.vue:421](webpack:///notifications/src/NotificationsApp.vue)
Started background fetcher as session_keepalive is enabled [NotificationsApp.vue:283](webpack:///notifications/src/NotificationsApp.vue)
Proxying an event bus of version 2.1.1 with 1.3.0 [index.es.js:2337:14](webpack:///nextcloud/node_modules/@nextcloud/auth/node_modules/@nextcloud/event-bus/dist/index.es.js)
Proxying an event bus of version 2.1.1 with 3.0.0 [index.js:19:12](webpack:///nextcloud/node_modules/@nextcloud/vue/node_modules/@nextcloud/event-bus/dist/index.js)
session heartbeat polling started [session-heartbeat.js:103:9](webpack:///nextcloud/core/src/session-heartbeat.js)
Proxying an event bus of version 2.1.1 with 3.0.2 [index.js:19:20](webpack://related_resources/node_modules/@nextcloud/axios/node_modules/@nextcloud/event-bus/dist/index.js)
Got notification data [NotificationsApp.vue:384](webpack:///notifications/src/NotificationsApp.vue)
Polling interval updated to 30000 [NotificationsApp.vue:421](webpack:///notifications/src/NotificationsApp.vue)
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 10 [globals.js:60:15](webpack:///nextcloud/core/src/globals.js)

```

**Additional context**
Last mention of anything with photo in it from the journal
```
Mar 25 11:34:28 nextcloud dbus-daemon[494091]: [session uid=1000 pid=494091] Activating via systemd: service name='org.gtk.vfs.GPhoto2VolumeMonitor' unit='gvfs-gphoto2-volume-monitor.service' requested by ':1.28' (uid=1000 pid=494271 comm="/usr/lib/x86_64-linux-gnu/tumbler-1/tumblerd " label="unconfined")
Mar 25 11:34:28 nextcloud dbus-daemon[494091]: [session uid=1000 pid=494091] Successfully activated service 'org.gtk.vfs.GPhoto2VolumeMonitor'
Mar 25 11:34:39 nextcloud systemd[494065]: gvfs-gphoto2-volume-monitor.service: Succeeded.
```

Last few nextcloud.log entries related to photo
```
{"reqId":"fyvjcEZJxnEVYl1sgHJf","level":3,"time":"2023-03-30T14:32:57+00:00","remoteAddr":"192.168.1.120","user":"xxxx","app":"index","method":"GET","url":"/index.php/apps/photos/api/v1/preview/12495739?x=64&y=64","message":"Could not create folder","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/111.0","version":"25.0.3.2","exception":{"Exception":"OCP\\Files\\NotPermittedException","Message":"Could not create folder","Code":0,"Trace":[{"file":"/var/www/html/lib/private/Files/AppData/AppData.php","line":147,"function":"newFolder","class":"OC\\Files\\Node\\Folder","type":"->"},{"file":"/var/www/html/lib/private/Preview/Storage/Root.php","line":74,"function":"newFolder","class":"OC\\Files\\AppData\\AppData","type":"->"},{"file":"/var/www/html/lib/private/Preview/Generator.php","line":581,"function":"newFolder","class":"OC\\Preview\\Storage\\Root","type":"->"},{"file":"/var/www/html/lib/private/Preview/Generator.php","line":137,"function":"getPreviewFolder","class":"OC\\Preview\\Generator","type":"->"},{"file":"/var/www/html/lib/private/Preview/Generator.php","line":114,"function":"generatePreviews","class":"OC\\Preview\\Generator","type":"->"},{"file":"/var/www/html/lib/private/PreviewManager.php","line":185,"function":"getPreview","class":"OC\\Preview\\Generator","type":"->"},{"file":"/var/www/html/apps/photos/lib/Controller/PreviewController.php","line":162,"function":"getPreview","class":"OC\\PreviewManager","type":"->"},{"file":"/var/www/html/apps/photos/lib/Controller/PreviewController.php","line":128,"function":"fetchPreview","class":"OCA\\Photos\\Controller\\PreviewController","type":"->"},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":225,"function":"index","class":"OCA\\Photos\\Controller\\PreviewController","type":"->"},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":133,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/lib/private/AppFramework/App.php","line":172,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/lib/private/Route/Router.php","line":298,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/html/lib/base.php","line":1047,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/html/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/html/lib/private/Files/Node/Folder.php","Line":166,"CustomMessage":"--"}}
{"reqId":"OTY3ogMo3SfN6tiESRu4","level":3,"time":"2023-03-30T14:32:58+00:00","remoteAddr":"192.168.1.120","user":"xxxx","app":"index","method":"GET","url":"/index.php/apps/photos/api/v1/preview/12495743?x=64&y=64","message":"Could not create folder","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/111.0","version":"25.0.3.2","exception":{"Exception":"OCP\\Files\\NotPermittedException","Message":"Could not create folder","Code":0,"Trace":[{"file":"/var/www/html/lib/private/Files/AppData/AppData.php","line":147,"function":"newFolder","class":"OC\\Files\\Node\\Folder","type":"->"},{"file":"/var/www/html/lib/private/Preview/Storage/Root.php","line":74,"function":"newFolder","class":"OC\\Files\\AppData\\AppData","type":"->"},{"file":"/var/www/html/lib/private/Preview/Generator.php","line":581,"function":"newFolder","class":"OC\\Preview\\Storage\\Root","type":"->"},{"file":"/var/www/html/lib/private/Preview/Generator.php","line":137,"function":"getPreviewFolder","class":"OC\\Preview\\Generator","type":"->"},{"file":"/var/www/html/lib/private/Preview/Generator.php","line":114,"function":"generatePreviews","class":"OC\\Preview\\Generator","type":"->"},{"file":"/var/www/html/lib/private/PreviewManager.php","line":185,"function":"getPreview","class":"OC\\Preview\\Generator","type":"->"},{"file":"/var/www/html/apps/photos/lib/Controller/PreviewController.php","line":162,"function":"getPreview","class":"OC\\PreviewManager","type":"->"},{"file":"/var/www/html/apps/photos/lib/Controller/PreviewController.php","line":128,"function":"fetchPreview","class":"OCA\\Photos\\Controller\\PreviewController","type":"->"},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":225,"function":"index","class":"OCA\\Photos\\Controller\\PreviewController","type":"->"},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":133,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/lib/private/AppFramework/App.php","line":172,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/lib/private/Route/Router.php","line":298,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/html/lib/base.php","line":1047,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/html/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/html/lib/private/Files/Node/Folder.php","Line":166,"CustomMessage":"--"}}
{"reqId":"VKkJY1JoKwitTn0CzvnW","level":3,"time":"2023-03-30T14:32:59+00:00","remoteAddr":"192.168.1.120","user":"xxxx","app":"index","method":"GET","url":"/index.php/apps/photos/api/v1/preview/12495574?x=512&y=512","message":"Could not create folder","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/111.0","version":"25.0.3.2","exception":{"Exception":"OCP\\Files\\NotPermittedException","Message":"Could not create folder","Code":0,"Trace":[{"file":"/var/www/html/lib/private/Files/AppData/AppData.php","line":147,"function":"newFolder","class":"OC\\Files\\Node\\Folder","type":"->"},{"file":"/var/www/html/lib/private/Preview/Storage/Root.php","line":74,"function":"newFolder","class":"OC\\Files\\AppData\\AppData","type":"->"},{"file":"/var/www/html/lib/private/Preview/Generator.php","line":581,"function":"newFolder","class":"OC\\Preview\\Storage\\Root","type":"->"},{"file":"/var/www/html/lib/private/Preview/Generator.php","line":137,"function":"getPreviewFolder","class":"OC\\Preview\\Generator","type":"->"},{"file":"/var/www/html/lib/private/Preview/Generator.php","line":114,"function":"generatePreviews","class":"OC\\Preview\\Generator","type":"->"},{"file":"/var/www/html/lib/private/PreviewManager.php","line":185,"function":"getPreview","class":"OC\\Preview\\Generator","type":"->"},{"file":"/var/www/html/apps/photos/lib/Controller/PreviewController.php","line":162,"function":"getPreview","class":"OC\\PreviewManager","type":"->"},{"file":"/var/www/html/apps/photos/lib/Controller/PreviewController.php","line":128,"function":"fetchPreview","class":"OCA\\Photos\\Controller\\PreviewController","type":"->"},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":225,"function":"index","class":"OCA\\Photos\\Controller\\PreviewController","type":"->"},{"file":"/var/www/html/lib/private/AppFramework/Http/Dispatcher.php","line":133,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/lib/private/AppFramework/App.php","line":172,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/lib/private/Route/Router.php","line":298,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/html/lib/base.php","line":1047,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/html/index.php","line":36,"function":"handleRequest","class":"OC","type":"::"}],"File":"/var/www/html/lib/private/Files/Node/Folder.php","Line":166,"CustomMessage":"--"}}
```

**Comment by ne20002** [comment]
I can confirm this. I just installed recognize and played a bit with the photos app. It definitely shows all pictures of a user, not only those inside the Photos folder. The configuration option is labled as folder for uploads, but it is unclear if this also shall be the base folder for photos app (if not, the app definitely needs a base folder option).

**Comment by gucu** [comment]
I can also confirm this but I am assuming that the default photos folder is simply to specify where you upload. I think we need an option to exclude or only include this folder, etc.
Thanks

**Comment by eXodussel1** [comment]
Hi, i have the same issue and it´s not only pictures from my own account that are shown but also from shared files which makes it more frustrating because i can´t create a .nomedia/.noimage file.

**Comment by joshtrichards** [comment]
Duplicate of #1567.

This is not a bug, but currently expected behavior. The referenced issue, however, tracks interest in an enhancement request. Please express your interest there with a :+1: if you'd like to see that functionality. You may also subscribe yourself to that issue to track progress.

Let's close this duplicate Issue please.

**Comment by limes007** [comment]
@joshtrichards the referenced issue is locked. We cannot express interest there.

**Comment by joshtrichards** [comment]
Fixed in #2319
